dissect/target/plugins/os/windows/defender.py

```diff
@@ -106,16 +407,114 @@ def evtx(self) -> Generator[Record, None, None]:
+    def recover(self, output_dir: Path) -> None:
+        """Recover files that have been placed into quarantine by Microsoft Defender.
 
-def filter_records(records: Iterable, field_name: str, field_value: Any) -> Generator[Record, None, None]:
-    def filter_func(record: Record) -> bool:
-        return hasattr(record, field_name) and getattr(record, field_name) == field_value
+        Microsoft Defender RC4 encrypts the output of the 'BackupRead' function when it places a file into quarantine.
+        This means multiple data streams can be contained in a single quarantined file, including zone identifier
+        information.
+        """
+        if not output_dir.exists():
+            raise ValueError("Output directory does not exist.")
+        quarantine_directory = self.target.fs.path(DEFENDER_QUARANTINE_DIR)
+        resourcedata_directory = quarantine_directory.joinpath("ResourceData")
+        if resourcedata_directory.exists() and resourcedata_directory.is_dir():
+            recovered_files = []
+            for entry in self.get_quarantine_entries():
+                for resource in entry.resources:
+                    if resource.detection_type != b"file":
+                        # We can only recover file entries
+                        continue
+                    # First two characters of the resource ID is the subdirectory that will contain the quarantined file
+                    subdir = resource.resource_id[0:2]
+                    resourcedata_location = resourcedata_directory.joinpath(subdir).joinpath(resource.resource_id)
+                    if not resourcedata_location.exists():
+                        self.target.log.warning(f"Could not find a ResourceData file for {entry.resource_id}.")
+                        continue
+                    if not resourcedata_location.is_file():
+                        self.target.log.warning(f"{resourcedata_location} is not a file!")
+                        continue
+                    if resourcedata_location in recovered_files:
+                        # We already recovered this file
+                        continue
+                    fh = resourcedata_location.open()
+                    # We restore the file with the resource_id as its filename. While we could 'guess' the filename
+                    # based on the information we have from the associated quarantine entry, there is a potential that
+                    # different files have the same filename. Analysts can use the quarantine records to cross
+                    # reference.
+                    for dest_filename, dest_buf in recover_quarantined_file_streams(fh, resource.resource_id):
+                        output_filename = output_dir.joinpath(dest_filename)
+                        self.target.log.info(f"Saving {output_filename}")
+                        with open(output_filename, "wb") as output_file:
+                            output_file.write(dest_buf)
```

Comment:
```suggestion
                        output_filename.write_bytes(resource_data_location.read_bytes())
```
You can also remove the `fh = resourcedata_location.open()` and `fh.close()` if you do this.